cli/config/addressbook.go

```diff
@@ -0,0 +1,158 @@
+			return fmt.Errorf("malformed address account name '%s': %w", name, err)
+		}
+
+		if err := acc.Validate(); err != nil {
+			return fmt.Errorf("address account '%s': %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+// Load loads the given address account.
+func (w *AddressBook) Load(name string) (*addressbook.Address, error) {
+	cfg, exists := w.All[name]
+	if !exists {
+		return nil, fmt.Errorf("address account '%s' does not exist in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return nil, fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	af := addressbook.AddressFactory{}
+	acc, err := af.Load(name, cfg.Config, cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make sure the address matches what we have in the config.
+	if expected, actual := cfg.GetAddress(), acc.Address(); !actual.Equal(expected) {
+		return nil, fmt.Errorf("address mismatch after loading account (expected: %s got: %s)",
+			expected,
+			actual,
+		)
+	}
+
+	return acc, nil
+}
+
+// Remove removes the given account.
+func (w *AddressBook) Remove(name string) error {
+	if _, exists := w.All[name]; !exists {
+		return fmt.Errorf("account '%s' does not exist in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	delete(w.All, name)
+
+	return nil
+}
+
+// Rename renames an existing account.
+func (w *AddressBook) Rename(old, new string) error {
+	cfg, exists := w.All[old]
+	if !exists {
+		return fmt.Errorf("account '%s' does not exist", old)
+	}
+
+	if _, exists = w.All[new]; exists {
+		return fmt.Errorf("account '%s' already exists", new)
+	}
+
+	if err := config.ValidateIdentifier(old); err != nil {
+		return fmt.Errorf("malformed old account name '%s': %w", old, err)
+	}
+	if err := config.ValidateIdentifier(new); err != nil {
+		return fmt.Errorf("malformed new account name '%s': %w", new, err)
+	}
+
+	w.All[new] = cfg
+	delete(w.All, old)
+
+	return nil
+}
+
+// Add imports an existing account.
+func (w *AddressBook) Add(name string, nw *AddressAccount, src *addressbook.ImportSource) error {
+	if _, exists := w.All[name]; exists {
+		return fmt.Errorf("account '%s' already exists in the address book", name)
+	}
+
+	if err := config.ValidateIdentifier(name); err != nil {
+		return fmt.Errorf("malformed account name '%s': %w", name, err)
+	}
+
+	af := addressbook.AddressFactory{}
+	acc, err := af.Add(nw.Config, src)
+	if err != nil {
+		return err
+	}
+
+	// Store address so we don't need to load the account each time to see the address.
+	address, err := acc.Address().MarshalText()
+	if err != nil {
+		return fmt.Errorf("failed to marshal account address: %w", err)
+	}
+	nw.Address = string(address)
+
+	if w.All == nil {
+		w.All = make(map[string]*AddressAccount)
+	}
+	w.All[name] = nw
+
+	return nil
+}
+
+// AddressAccount is an address account configuration object.
```

Comment:
As elsewhere this should be called either just an `Address` or `NamedAddress` or `AddressBookEntry` etc. as it is not an account.